sql/catalyst/src/main/scala/org/apache/spark/sql/BaseRow.java

```diff
@@ -149,6 +149,24 @@ public int fieldIndex(String name) {
   }
 
   @Override
+  public boolean equals(Object other) {
```

Comment:
This is the default implementation, could be used to check equals between different implementation  of Row, for example, MutableRow() == SpecificRow()